base_layer/core/tests/mempool.rs

```diff
@@ -397,6 +397,13 @@ fn test_reorg() {
     assert_eq!(stats.unconfirmed_txs, 2);
     assert_eq!(stats.timelocked_txs, 1);
     assert_eq!(stats.published_txs, 3);
+
+    // test that process_reorg can handle the case when removed_blocks is empty
+    // see https://github.com/tari-project/tari/issues/2101#issuecomment-680726940
+    let template = chain_block(&blocks[3], vec![], consensus_manager.consensus_constants());
+    let reorg_block4 = db.calculate_mmr_roots(template).unwrap();
+
+    assert_eq!(mempool.process_reorg(vec![], vec![reorg_block4]).unwrap(), ());
```

Comment:
Out of interest, this would also be ok, since a panic would fail the test
```suggestion
    mempool.process_reorg(vec![], vec![reorg_block4]).unwrap();
```

Reply:
Yeah I did previously have it that way but wanted to make it more explicit. Will revert